charts/aws-fsx-csi-driver/templates/controller-deployment.yaml

```diff
@@ -0,0 +1,105 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: fsx-csi-controller
+  labels:
+    {{- include "aws-fsx-csi-driver.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.controller.replicaCount }}
+  selector:
+    matchLabels:
+      app: fsx-csi-controller
```

Comment:
I'll test out the upgrade case, since I am bumping major version of helm chart I am not sure without testing how helm deals with it. I just cargo culted how ebs does it TBH: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/charts/aws-ebs-csi-driver/templates/controller.yaml#L12

Reply:
this change is safe because helm somehow knows to delete and recreate the deployment which also deletes and recreates the pods..

1. I installed the current helm chart v0.1.0 using the README command 
  a. (I manually edited the deployment and daemonset to have hostNetwork and to point to image 0.4.0 instead of 0.3.0 because they were crashlooping as I have instance metadata disabled)
2. I ran `helm upgrade --install aws-fsx-csi-driver  ./charts/aws-fsx-csi-driver`
3. helm recreated the deployment
```
k get deployment
NAME                 READY   UP-TO-DATE   AVAILABLE   AGE
fsx-csi-controller   1/1     1            1           57s
```

Reply:
I didn't notice that the name of the deployment was changing.  That explains the delete and create behavior.